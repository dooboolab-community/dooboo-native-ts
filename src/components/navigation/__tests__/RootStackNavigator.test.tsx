import 'react-native';

import React, { ReactElement } from 'react';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import StackNavigator from '../RootStackNavigator';
import renderer from 'react-test-renderer';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;

describe('[Stack] navigator', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(
      <StackNavigator {...props} />,
    );
  });

  it('should renders without crashing', () => {
    jest.useFakeTimers();
    renderer.create(component).toJSON();
    jest.runAllTimers();
    // expect(rendered).toMatchSnapshot();
    // expect(rendered).toBeTruthy();
  });
});
