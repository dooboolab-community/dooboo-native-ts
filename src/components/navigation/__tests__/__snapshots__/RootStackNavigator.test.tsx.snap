// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Stack] navigator should renders without crashing 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={true}
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 64,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "backgroundColor": "white",
                              "flexBasis": 0,
                              "flexDirection": "column",
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "justifyContent": "flex-start",
                              "overflow": "hidden",
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "column",
                                "height": "100%",
                                "justifyContent": "flex-start",
                                "width": "100%",
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "black",
                                  "fontSize": 18,
                                  "lineHeight": 27,
                                },
                                Object {
                                  "marginTop": 100,
                                },
                              ]
                            }
                          >
                            
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "black",
                                  "fontSize": 18,
                                  "lineHeight": 27,
                                },
                              ]
                            }
                          >
                            
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "black",
                                  "fontSize": 18,
                                  "lineHeight": 27,
                                },
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "alignSelf": "center",
                                "bottom": 40,
                                "flexDirection": "column",
                                "position": "absolute",
                                "width": "85%",
                              },
                            ]
                          }
                        >
                          <TouchableOpacity
                            activeOpacity={0.5}
                            testID="btn-login"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "backgroundColor": "#069ccd",
                                    "borderColor": "#069ccd",
                                    "borderRadius": 4,
                                    "borderWidth": 2,
                                    "height": 52,
                                    "justifyContent": "center",
                                    "width": 320,
                                  },
                                ]
                              }
                            >
                              <Image
                                source={
                                  Object {
                                    "canInstrument": true,
                                    "createTransformer": [Function],
                                    "getCacheKey": [Function],
                                    "process": [Function],
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "height": 24,
                                      "left": 16,
                                      "position": "absolute",
                                      "width": 24,
                                    },
                                  ]
                                }
                              />
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "white",
                                      "fontSize": 14,
                                    },
                                  ]
                                }
                              >
                                Login
                              </Text>
                            </View>
                          </TouchableOpacity>
                          <View
                            style={
                              Object {
                                "marginTop": 8,
                              }
                            }
                          />
                          <TouchableOpacity
                            activeOpacity={0.5}
                            testID="btn-navigate"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "backgroundColor": "#069ccd",
                                    "borderColor": "#069ccd",
                                    "borderRadius": 4,
                                    "borderWidth": 2,
                                    "height": 52,
                                    "justifyContent": "center",
                                    "width": 320,
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "white",
                                      "fontSize": 14,
                                    },
                                  ]
                                }
                              >
                                Navigate to Temp
                              </Text>
                            </View>
                          </TouchableOpacity>
                          <View
                            style={
                              Object {
                                "marginTop": 8,
                              }
                            }
                          />
                          <TouchableOpacity
                            activeOpacity={0.5}
                            testID="btn-theme"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "backgroundColor": "#069ccd",
                                    "borderColor": "#069ccd",
                                    "borderRadius": 4,
                                    "borderWidth": 2,
                                    "height": 52,
                                    "justifyContent": "center",
                                    "width": 320,
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "white",
                                      "fontSize": 14,
                                    },
                                  ]
                                }
                              >
                                Change theme
                              </Text>
                            </View>
                          </TouchableOpacity>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          pointerEvents="box-none"
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderBottomColor": "rgb(199, 199, 204)",
                  "flex": 1,
                  "shadowColor": "rgb(199, 199, 204)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 64,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 20,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "marginHorizontal": 18,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Intro
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</View>
`;
